CDP4Dal/Session.cs

```diff
@@ -729,25 +750,58 @@ public async Task Reload()
         /// <summary>
         /// Can a Cancel action be executed?
         /// </summary>
-        /// <returns>True is Cancel is allowed, otherwise false.</returns>
-        public bool CanCancel()
+        /// <param name="cancellationTokenSource">
+        /// Cancellation token source that will be cancelled <see cref="CancellationTokenSource"/>
+        /// </param>
+        /// <returns>
+        /// True if Cancel is allowed, otherwise false.
+        /// </returns>
+        private static bool CanCancel(CancellationTokenSource cancellationTokenSource)
         {
-            if (this.cancellationTokenSource == null)
+            if (cancellationTokenSource == null)
             {
                 return false;
             }
 
-            return this.cancellationTokenSource.Token.CanBeCanceled && !this.cancellationTokenSource.IsCancellationRequested;
+            return cancellationTokenSource.Token.CanBeCanceled && !cancellationTokenSource.IsCancellationRequested;
+        }
+
+        /// <summary>
+        /// Can a Cancel action be executed?
+        /// </summary>
+        /// <returns>
+        /// True if Cancel is allowed for at least on token, otherwise false.
+        /// </returns>
+        public bool CanCancel()

```

Comment:
this method needs to be added to the `ISession` interface

Reply:
it already has been.